From 86bf0019ed9c0d1e19916448454c3f7df0479ac6 Mon Sep 17 00:00:00 2001
From: Joseph Myers <joseph@codesourcery.com>
Date: Mon, 18 Jun 2018 13:34:52 +0000
Subject: [PATCH] Add MSG_STAT_ANY from Linux 4.17 to bits/msq.h.

Linux 4.17 adds a MSG_STAT_ANY constant (ipcs command).  This patch
adds it to the relevant bits/msq.h headers.

Tested for x86_64.

	* sysdeps/unix/sysv/linux/alpha/bits/msq.h [__USE_MISC]
	(MSG_STAT_ANY): New macro.
	* sysdeps/unix/sysv/linux/bits/msq.h [__USE_MISC]
	(MSG_STAT_ANY): Likewise.
	* sysdeps/unix/sysv/linux/generic/bits/msq.h [__USE_MISC]
	(MSG_STAT_ANY): Likewise.
	* sysdeps/unix/sysv/linux/hppa/bits/msq.h [__USE_MISC]
	(MSG_STAT_ANY): Likewise.
	* sysdeps/unix/sysv/linux/ia64/bits/msq.h [__USE_MISC]
	(MSG_STAT_ANY): Likewise.
	* sysdeps/unix/sysv/linux/mips/bits/msq.h [__USE_MISC]
	(MSG_STAT_ANY): Likewise.
	* sysdeps/unix/sysv/linux/powerpc/bits/msq.h [__USE_MISC]
	(MSG_STAT_ANY): Likewise.
	* sysdeps/unix/sysv/linux/s390/bits/msq.h [__USE_MISC]
	(MSG_STAT_ANY): Likewise.
	* sysdeps/unix/sysv/linux/sparc/bits/msq.h [__USE_MISC]
	(MSG_STAT_ANY): Likewise.
	* sysdeps/unix/sysv/linux/x86/bits/msq.h [__USE_MISC]
	(MSG_STAT_ANY): Likewise.
---
 ChangeLog                                  | 21 +++++++++++++++++++++
 sysdeps/unix/sysv/linux/alpha/bits/msq.h   |  1 +
 sysdeps/unix/sysv/linux/bits/msq.h         |  1 +
 sysdeps/unix/sysv/linux/generic/bits/msq.h |  1 +
 sysdeps/unix/sysv/linux/hppa/bits/msq.h    |  1 +
 sysdeps/unix/sysv/linux/ia64/bits/msq.h    |  1 +
 sysdeps/unix/sysv/linux/mips/bits/msq.h    |  1 +
 sysdeps/unix/sysv/linux/powerpc/bits/msq.h |  1 +
 sysdeps/unix/sysv/linux/s390/bits/msq.h    |  1 +
 sysdeps/unix/sysv/linux/sparc/bits/msq.h   |  1 +
 sysdeps/unix/sysv/linux/x86/bits/msq.h     |  1 +
 11 files changed, 31 insertions(+)

Index: b/sysdeps/unix/sysv/linux/alpha/bits/msq.h
===================================================================
--- a/sysdeps/unix/sysv/linux/alpha/bits/msq.h
+++ b/sysdeps/unix/sysv/linux/alpha/bits/msq.h
@@ -56,6 +56,7 @@ struct msqid_ds
 /* ipcs ctl commands */
 # define MSG_STAT 11
 # define MSG_INFO 12
+# define MSG_STAT_ANY 13
 
 /* buffer for msgctl calls IPC_INFO, MSG_INFO */
 struct msginfo
Index: b/sysdeps/unix/sysv/linux/bits/msq.h
===================================================================
--- a/sysdeps/unix/sysv/linux/bits/msq.h
+++ b/sysdeps/unix/sysv/linux/bits/msq.h
@@ -59,6 +59,7 @@ struct msqid_ds
 /* ipcs ctl commands */
 # define MSG_STAT 11
 # define MSG_INFO 12
+# define MSG_STAT_ANY 13
 
 /* buffer for msgctl calls IPC_INFO, MSG_INFO */
 struct msginfo
Index: b/sysdeps/unix/sysv/linux/generic/bits/msq.h
===================================================================
--- a/sysdeps/unix/sysv/linux/generic/bits/msq.h
+++ b/sysdeps/unix/sysv/linux/generic/bits/msq.h
@@ -66,6 +66,7 @@ struct msqid_ds
 /* ipcs ctl commands */
 # define MSG_STAT 11
 # define MSG_INFO 12
+# define MSG_STAT_ANY 13
 
 /* buffer for msgctl calls IPC_INFO, MSG_INFO */
 struct msginfo
Index: b/sysdeps/unix/sysv/linux/hppa/bits/msq.h
===================================================================
--- a/sysdeps/unix/sysv/linux/hppa/bits/msq.h
+++ b/sysdeps/unix/sysv/linux/hppa/bits/msq.h
@@ -66,6 +66,7 @@ struct msqid_ds
 /* ipcs ctl commands */
 # define MSG_STAT 11
 # define MSG_INFO 12
+# define MSG_STAT_ANY 13
 
 /* buffer for msgctl calls IPC_INFO, MSG_INFO */
 struct msginfo
Index: b/sysdeps/unix/sysv/linux/ia64/bits/msq.h
===================================================================
--- a/sysdeps/unix/sysv/linux/ia64/bits/msq.h
+++ b/sysdeps/unix/sysv/linux/ia64/bits/msq.h
@@ -51,6 +51,7 @@ struct msqid_ds
 /* ipcs ctl commands */
 # define MSG_STAT 11
 # define MSG_INFO 12
+# define MSG_STAT_ANY 13
 
 /* buffer for msgctl calls IPC_INFO, MSG_INFO */
 struct msginfo
Index: b/sysdeps/unix/sysv/linux/mips/bits/msq.h
===================================================================
--- a/sysdeps/unix/sysv/linux/mips/bits/msq.h
+++ b/sysdeps/unix/sysv/linux/mips/bits/msq.h
@@ -74,6 +74,7 @@ struct msqid_ds
 /* ipcs ctl commands */
 # define MSG_STAT 11
 # define MSG_INFO 12
+# define MSG_STAT_ANY 13
 
 /* buffer for msgctl calls IPC_INFO, MSG_INFO */
 struct msginfo
Index: b/sysdeps/unix/sysv/linux/powerpc/bits/msq.h
===================================================================
--- a/sysdeps/unix/sysv/linux/powerpc/bits/msq.h
+++ b/sysdeps/unix/sysv/linux/powerpc/bits/msq.h
@@ -65,6 +65,7 @@ struct msqid_ds
 /* ipcs ctl commands */
 # define MSG_STAT 11
 # define MSG_INFO 12
+# define MSG_STAT_ANY 13
 
 /* buffer for msgctl calls IPC_INFO, MSG_INFO */
 struct msginfo
Index: b/sysdeps/unix/sysv/linux/s390/bits/msq.h
===================================================================
--- a/sysdeps/unix/sysv/linux/s390/bits/msq.h
+++ b/sysdeps/unix/sysv/linux/s390/bits/msq.h
@@ -66,6 +66,7 @@ struct msqid_ds
 /* ipcs ctl commands */
 # define MSG_STAT 11
 # define MSG_INFO 12
+# define MSG_STAT_ANY 13
 
 /* buffer for msgctl calls IPC_INFO, MSG_INFO */
 struct msginfo
Index: b/sysdeps/unix/sysv/linux/sparc/bits/msq.h
===================================================================
--- a/sysdeps/unix/sysv/linux/sparc/bits/msq.h
+++ b/sysdeps/unix/sysv/linux/sparc/bits/msq.h
@@ -66,6 +66,7 @@ struct msqid_ds
 /* ipcs ctl commands */
 # define MSG_STAT 11
 # define MSG_INFO 12
+# define MSG_STAT_ANY 13
 
 /* buffer for msgctl calls IPC_INFO, MSG_INFO */
 struct msginfo
Index: b/sysdeps/unix/sysv/linux/x86/bits/msq.h
===================================================================
--- a/sysdeps/unix/sysv/linux/x86/bits/msq.h
+++ b/sysdeps/unix/sysv/linux/x86/bits/msq.h
@@ -64,6 +64,7 @@ struct msqid_ds
 /* ipcs ctl commands */
 # define MSG_STAT 11
 # define MSG_INFO 12
+# define MSG_STAT_ANY 13
 
 /* buffer for msgctl calls IPC_INFO, MSG_INFO */
 struct msginfo
